### PR TITLE
CSP-805 adding start and end date parameters to GetCalenderEvents

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.AANHub.Domain/Interfaces/Repositories/ICalendarEventsReadRepository.cs [hunk 1]
```diff
@@ -5,6 +5,6 @@ namespace SFA.DAS.AANHub.Domain.Interfaces.Repositories;
 
 public interface ICalendarEventsReadRepository
 {
-    Task<List<CalendarEventSummary>> GetCalendarEvents(Guid memberId, CancellationToken cancellationToken);
+    Task<List<CalendarEventSummary>> GetCalendarEvents(Guid memberId, DateTime? startDate, DateTime? endDate, CancellationToken cancellationToken);
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Domain/Interfaces/Repositories/ICalendarEventsReadRepository.cs:8]: Make dates non nullable here.

File: src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs [hunk 2]
```diff
@@ -24,8 +24,12 @@ await _aanDataContext
 			.Include(x => x.Calendar)
 			.SingleOrDefaultAsync();
 
-	public async Task<List<CalendarEventSummary>> GetCalendarEvents(Guid memberId, CancellationToken cancellationToken)
+	public async Task<List<CalendarEventSummary>> GetCalendarEvents(Guid memberId, DateTime? startDate, DateTime? endDate, CancellationToken cancellationToken)
 	{
+		startDate ??= DateTime.Today;
+
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs:30]: that will happen anyway

File: src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs [hunk 3]
```diff
@@ -24,8 +24,12 @@ await _aanDataContext
 			.Include(x => x.Calendar)
 			.SingleOrDefaultAsync();
 
-	public async Task<List<CalendarEventSummary>> GetCalendarEvents(Guid memberId, CancellationToken cancellationToken)
+	public async Task<List<CalendarEventSummary>> GetCalendarEvents(Guid memberId, DateTime? startDate, DateTime? endDate, CancellationToken cancellationToken)
 	{
+		startDate ??= DateTime.Today;
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs:29]: ok good catch

File: src/SFA.DAS.AANHub.Domain/Interfaces/Repositories/ICalendarEventsReadRepository.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.AANHub.Domain/Interfaces/Repositories/ICalendarEventsReadRepository.cs:8]: ok

File: src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs:30]: oh I see, not even touch the database. Ok

File: src/SFA.DAS.AANHub.Application/CalendarEvents/Queries/GetCalendarEventsQueryHandler.cs [hunk 4]
```diff
@@ -15,15 +16,32 @@ public GetCalendarEventsQueryHandler(ICalendarEventsReadRepository calendarEvent
 
     public async Task<ValidatedResponse<GetCalendarEventsQueryResult>> Handle(GetCalendarEventsQuery request, CancellationToken cancellationToken)
     {
-        var pageSize = Domain.Common.Constants.CalendarEvents.PageSize;
+        var pageSize = 0;
         var page = 1;
+        var startDate = request.StartDate ?? DateTime.Today;
+
+        if (startDate < DateTime.Today)
+            startDate = DateTime.Today;
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/CalendarEvents/Queries/GetCalendarEventsQueryHandler.cs:24]: var startDate = request.StartDate == null || request.StartDate < DateTime.Today ? DateTime.Today : request.StartDate;

File: src/SFA.DAS.AANHub.Application/CalendarEvents/Queries/GetCalendarEventsQueryHandler.cs [hunk 5]
```diff
@@ -15,15 +16,32 @@ public GetCalendarEventsQueryHandler(ICalendarEventsReadRepository calendarEvent
...
+
+        var endDate = request.EndDate ?? DateTime.Today.AddYears(1);
+
+        if (startDate > endDate)
+            return new ValidatedResponse<GetCalendarEventsQueryResult>(
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/CalendarEvents/Queries/GetCalendarEventsQueryHandler.cs:29]: If `if` is not completed in the same line, it has to have a block

**Review by DevenKShah — DISMISSED** [review]
Couple of minor suggestions

**Pull request merged**